### PR TITLE
fix memory leak in c on windows

### PR DESCRIPTION
**Pull request by yunhaoling**
1. **`result->addrInfo->ai_addr = calloc(1, sizeof(struct sockaddr_in))) == NULL` should no longer exist according to cuamqp
echoing the change in PR: https://github.com/Azure/azure-c-shared-utility/pull/515**

----------------------

2. **The PR to fix the following memory leak issue has been merged already: https://github.com/Azure/azure-c-shared-utility/pull/540**

- output buffer should be freed manually

https://github.com/Azure/azure-uamqp-python/blob/4558df974fa1a483cf5f0eaf7fe32af465292a50/src/vendor/azure-uamqp-c/deps/azure-c-shared-utility/adapters/tlsio_schannel.c#L298

by:

```c
        FreeContextBuffer(init_security_buffers[0].pvBuffer);
        FreeContextBuffer(init_security_buffers[1].pvBuffer);
```

because the flag `ISC_REQ_ALLOCATE_MEMORY` is used in code:
https://github.com/Azure/azure-uamqp-python/blob/4558df974fa1a483cf5f0eaf7fe32af465292a50/src/vendor/azure-uamqp-c/deps/azure-c-shared-utility/adapters/tlsio_schannel.c#L365-L368
```
ISC_REQ_ALLOCATE_MEMORY | The [*security package*](../secgloss/s-gly.md) allocates output buffers for you. When you have finished using the output buffers, free them by calling the [FreeContextBuffer](/windows/win32/api/sspi/nf-sspi-freecontextbuffer) function.
```

according to
https://docs.microsoft.com/en-us/windows/win32/secauthn/initializesecuritycontext--schannel

### CODE REVIEW

**Comment by yunhaoling** [comment]
created PR in C repo: https://github.com/Azure/azure-c-shared-utility/pull/540

File: src/vendor/azure-uamqp-c/deps/azure-c-shared-utility/adapters/tlsio_schannel.c [hunk 1]
```diff
@@ -344,6 +344,7 @@ static void send_client_hello(TLS_IO_INSTANCE* tls_io_instance)
     {
         tls_io_instance->tlsio_state = TLSIO_STATE_ERROR;
         indicate_error(tls_io_instance);
+        (void)FreeCredentialHandle(&tls_io_instance->credential_handle);
```
Comment (yunhaoling) [inline src/vendor/azure-uamqp-c/deps/azure-c-shared-utility/adapters/tlsio_schannel.c:347]:
```suggestion
```

File: HISTORY.rst [hunk 2]
```diff
@@ -3,6 +3,10 @@
 Release History
 ===============
 
+1.4.2 (Unreleased)
+
+- Fiexed memory leak in win32 socketio (azure-sdk-for-python issue #19777).
```
Comment (yunhaoling) [inline HISTORY.rst:8]:
```suggestion
- Fixed memory leak in win32 tlsio (azure-sdk-for-python issue #19777).
```

File: HISTORY.rst [hunk 3]
```diff
@@ -3,6 +3,10 @@
 Release History
 ===============
 
+1.4.2 (Unreleased)
+
+- Fixed memory leak in win32 tlsio (azure-sdk-for-python issue #19777).
```
Comment (yunhaoling) [inline HISTORY.rst:8]:
```suggestion
- Fixed memory leak in win32 socketio and tlsio (azure-sdk-for-python issue #19777).
```

**Review by annatisch — APPROVED** [review]
I honestly have no idea on this one - so will trust you :)
Let's just make sure the long running stress tests don't reveal any issues.

**Pull request merged**